tests/baselines/reference/conditionalTypes2.types

```diff
@@ -403,7 +403,7 @@ interface B1<T> extends A1<T> {
 >T : T
 
     boom: T extends any ? true : true
->boom : T extends any ? true : true

```

Comment:
why did this type collapse to `true`?

Reply:
I changed this to eagerly resolve to `true` when the extends type is `any` or `unknown` (since it always will be).

Reply:
Ah, but what about `T extends any ? { x: T } : never` - that needs to distribute.

Reply:
Hmm, yes, I suppose we can only optimize when the conditional type isn't distributive.

Reply:
There's a better check (than just distributivity) that I have an outstanding PR for: instantiate with wildcard. If the instantiation isn't the input type, you can't simplify. If it is, you can.